// @switchboard-xyz/common-legacy/dist/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  LegacyCrossbarClient: () => LegacyCrossbarClient
});
module.exports = __toCommonJS(index_exports);
var import_common = require("@switchboard-xyz/common");
var import_axios = __toESM(require("axios"), 1);
var import_bs58 = __toESM(require("bs58"), 1);
var LegacyCrossbarClient = class {
  crossbarUrl;
  verbose;
  // feed hash -> crossbar response (v1 format)
  feedCache = /* @__PURE__ */ new Map();
  constructor(crossbarUrl, verbose) {
    this.crossbarUrl = new URL(crossbarUrl).origin;
    this.verbose = !!verbose;
  }
  /**
   * GET /fetch/:feedHash
   * Fetch data from the crossbar using the provided feedHash (v1 legacy format)
   * @deprecated Use fetchOracleFeed() from @switchboard-xyz/common instead.
   * @param {string} feedHash - The hash of the feed to fetch data for
   * @returns {Promise<CrossbarFetchResponse>} - The data fetched from the crossbar
   */
  async fetch(feedHash) {
    try {
      const cached = this.feedCache.get(feedHash);
      if (cached) return cached;
      const response = await import_axios.default.get(`${this.crossbarUrl}/fetch/${feedHash}`).then((resp) => resp.data);
      this.feedCache.set(feedHash, response);
      return response;
    } catch (err) {
      if (!import_axios.default.isAxiosError(err)) throw err;
      const response = err.response;
      if (!response) throw err;
      if (this.verbose) console.error(`${response.status}: ${response.data}`);
      throw new Error(`Bad Crossbar fetch status: ${response.status}`);
    }
  }
  /**
   * POST /store
   * Store oracle jobs on the crossbar, associated with a queue address (v1 legacy format)
   * @deprecated Use storeOracleFeed() from @switchboard-xyz/common instead.
   * @param {string} queueAddress - The address of the queue
   * @param {IOracleJob[]} jobs - The oracle jobs to store
   * @returns {Promise<{ cid: string; feedHash: string; queueHex: string }>} - The stored data information
   */
  async store(queueAddress, jobs) {
    try {
      const queue = (0, import_common.decodeString)(queueAddress);
      if (!queue) throw new Error(`Unable to parse queue: ${queueAddress}`);
      return await import_axios.default.post(
        `${this.crossbarUrl}/store`,
        { queue: import_bs58.default.encode(queue), jobs },
        { headers: { "Content-Type": "application/json" } }
      ).then((resp) => {
        if (resp.status === 200) return resp.data;
        throw new Error(`Bad Crossbar store response: ${resp.status}`);
      });
    } catch (err) {
      if (!import_axios.default.isAxiosError(err)) throw err;
      const response = err.response;
      if (!response) throw err;
      if (this.verbose) console.error(`${response.status}: ${response.data}`);
      throw new Error(`Bad Crossbar store response: ${response.status}`);
    }
  }
  /**
   * GET /simulate/:feedHash
   * Simulate a feed using the legacy v1 endpoint
   * @deprecated Use simulateFeed() from @switchboard-xyz/common instead.
   * @param {string} feedHash - The hash of the feed to simulate
   * @param {boolean} [includeReceipts] - Whether to include receipts in the response
   * @returns {Promise<{ feedHash: string; results: string[]; receipts?: any[] }[]>} - The simulation results
   */
  async simulate(feedHash, includeReceipts) {
    try {
      const params = includeReceipts ? { includeReceipts: true } : {};
      return await import_axios.default.get(`${this.crossbarUrl}/simulate/${feedHash}`, { params }).then((resp) => resp.data);
    } catch (err) {
      if (!import_axios.default.isAxiosError(err)) throw err;
      const response = err.response;
      if (!response) throw err;
      if (this.verbose) console.error(`${response.status}: ${response.data}`);
      throw new Error(`Bad Crossbar simulate status: ${response.status}`);
    }
  }
  /**
   * POST /gateways/fetch_signatures
   * Fetch signatures from oracles for a given set of jobs (v1 legacy format)
   * @deprecated Use fetchSignaturesConsensus() from @switchboard-xyz/common instead.
   * @param {FetchSignaturesRequest} request - The request parameters
   * @param {string} [network] - Optional network parameter (devnet/mainnet). Defaults to mainnet.
   * @returns {Promise<{ responses: FeedEvalResponse[]; failures: string[] }>} - Array of oracle signatures and results
   */
  async fetchSignatures(request, network) {
    try {
      return await import_axios.default.post(`${this.crossbarUrl}/gateways/fetch_signatures`, request, {
        params: { network }
      }).then((resp) => resp.data);
    } catch (err) {
      if (!import_axios.default.isAxiosError(err)) throw err;
      const response = err.response;
      if (!response) throw err;
      if (this.verbose) console.error(`${response.status}: ${response.data}`);
      throw new Error(
        `Bad Crossbar fetchOracleSignatures response: ${response.status}`
      );
    }
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LegacyCrossbarClient
});
